ui/src/components/common/NotificationCenter.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import Page from '../common/Page'
+import PageSection from './PageSection'
+import { useRelationshipRequestNotifications } from '../common/RelationshipRequestNotifications'
+import { useCCPCustomerNotifications } from '../CCP/CCPCustomerNotifications'
+import { useBrokerCustomerInviteNotifications } from '../Investor/BrokerCustomerInviteNotifications'
+import { useCCPCustomerInviteNotifications } from '../Investor/CCPInviteNotifications'
+import { useDismissibleNotifications } from '../common/DismissibleNotifications'
+import { useExchangeInviteNotifications } from '../Investor/ExchangeInviteNotifications'
+
+import { NotificationCenterIcon, ArrowLeftIcon } from '../../icons/Icons'
+
+type Props = {
+  sideNav: React.ReactElement;
+  onLogout: () => void;
+  showNotificationAlert?: boolean;
+  handleNotificationAlert?: () => void;
+}
+
+export const useAllNotifications = () => {
+  const notifications = [
+    ...useRelationshipRequestNotifications(),
+    ...useCCPCustomerNotifications(),
+    ...useCCPCustomerInviteNotifications(),
+    ...useBrokerCustomerInviteNotifications(),
+    ...useExchangeInviteNotifications(),
+    ...useDismissibleNotifications()
+  ];
+
+  return notifications;
+}
+
+const NotificationCenter: React.FC<Props> = ({ sideNav, onLogout, showNotificationAlert, handleNotificationAlert }) => {
+  const history = useHistory();
+  const notifications = useAllNotifications();
+  const [prevLocation, setPrevLocation] = useState<string>();
+  const [prevPath, setPrevPath] = useState<string>();
+
+  const historyState = history.location.state as string;
+
+  useEffect(() => {
+    const tempLocation = historyState?.split('/');
+    const tempPath = historyState;
+    if (historyState) {
+      localStorage.setItem('prevLocation', tempLocation[tempLocation.length - 1]);
+      localStorage.setItem('prevPath', tempPath);
+      setPrevLocation(tempLocation[tempLocation.length - 1]);
+      setPrevPath(tempPath);
+    } else {
+      const tempLocation = localStorage.getItem('prevLocation');
+      const tempPrevPath = localStorage.getItem('prevPath');
+      if (tempLocation) setPrevLocation(tempLocation);
+      if(tempPrevPath) setPrevPath(tempPrevPath);
+    }
+  });
+
+  const handleGoBack = () => {
+      history.push(`${prevPath}`);
+  }
```

Comment:
Have you tried `history.goBack()` instead of `history.push(prevPath)`? The former is built-in to the react-router library, and should allow you to remove the whole useEffect above that tracks and stores the previous page. 

To get the name of the back page for the `Return to ...` text, I think it should work to pass that in as a prop instead.

Reply:
Initially it was history.goBack but then on refresh it lost all the data and the functionality didn't work. So i tried to work around it and that was the only way i found how to do so. If you have any other ideas I'm more then happy to try them out.